docs/topics/db/queries.txt

```diff
@@ -1009,6 +1009,63 @@ Unless you are sure you wish to work with SQL ``NULL`` values, consider setting
 Key, index, and path transforms
 -------------------------------
 
+.. class:: KeyTransform(key_name, *args, **kwargs)
+
+A :class:`~django.db.models.Transform` to access a key named ``key_name`` in a
+:class:`~django.db.models.JSONField`. ``key_name`` could also be an integer to
+transform to an array index. The field instance which is to be transformed
+should be passed as the second argument. This would return the resulting key
+transform which is also in the form of json.
+
+To query on a :class:`~django.db.models.JSONField`::
+
+    >>> Dog.objects.create(name='Rufus', data={
+    ...     'breed': 'labrador',
+    ...     'owner': {
+    ...         'name': 'Bob',
+    ...         'other_pets': [{
+    ...             'name': 'Fishy',
+    ...         }],
+    ...     },
+    ... })
+    <Dog: Rufus>
+    >>> Dog.objects.annotate(breed_name=KeyTransform('breed', 'data')).filter(
+    ...     breed_name__exact='labrador'
+    ... )
+    <QuerySet [<Dog: Rufus>]>
+
+Nested key transforms can also be done as follows::
+
+    >>> Dogs.objects.annotate(
+    ...     owner_other_pet_name=KeyTransform(
+    ...         'other_pets',KeyTransform('owner', 'data')
+    ...         )).filter(owner_other_pet_name__exact=[{'name': 'Fishy'}])
+    <QuerySet [<Dog: Rufus>]>
+
+To query the resulting json as text, the .. class:: Cast function needs to be
```

Comment:
`.. class:: Cast` is not a valid directive in Sphinx.

Reply:
Please fix all invalid directives, e.g. `.. class:: json.KeyTextTransform` or `.. class:: json.KeyTransform`.